src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.File.Create.cs

```diff
@@ -26,6 +27,41 @@ public void FileSystemWatcher_File_Create()
             }
         }
 
+        [OuterLoop]
+        [Fact]
+        public void FileSystemWatcher_File_Create_MultipleWatchers_ExecutionContextFlowed()
+        {
+            ExecuteWithRetry(() =>
```

Comment:
Why are we retrying this ? 
And if retrying is necessary cant we just retry 
```C#
File.Create(fileName).Dispose();
Task.WaitAll(new[] { tcs1.Task, tcs2.Task }, WaitForExpectedEventTimeout);
Assert.Equal(42, tcs1.Task.Result);
Assert.Equal(84, tcs2.Task.Result);
```

Reply:
> Why are we retrying this ?

The FileSystemWatcher tests have been notoriously flaky because the underlying OS support is a bit flaky, subject to machine load and the like.  So we retry in pretty much all of our FileSystemWatcher tests.  All of the tests today use a retry buried in the helpers being used, but those helpers only target one watcher, and I needed support for two watchers, so I added this general ExecuteWithRetry to make that case easier.

> And if retrying is necessary cant we just retry

I wanted to avoid any complications from, for example, one watcher having completed due to the previous change and setting its TCS, and then the next watcher completing for the subsequent change and setting its TCS.  I thought it better to just retry the whole thing as a result.  That's also what we do in the existing retries.